[ffmpeg] build failure on arm64-osx

**Host Environment**

GitHub workflow: https://github.com/daschuer/vcpkg/actions/runs/4588810226

**To Reproduce**

run the workflow 

`./vcpkg install ffmpeg`

**Failure logs**

ffmpeg 5.0.2: 
https://github.com/microsoft/vcpkg/commit/47a020d619021539a2533e56ef850696065b23a5
[build-arm64-osx-min1100-dbg-config-47a020d.log](https://github.com/microsoft/vcpkg/files/11146121/build-arm64-osx-min1100-dbg-config-47a020d.log)

ffmpeg 5.1.2 after reordering ./configure parameters
https://github.com/microsoft/vcpkg/commit/37fce29a263a1546262eaed6af7ccac53b0094cd
[build-arm64-osx-min1100-rel-config-37fce29a.log](https://github.com/microsoft/vcpkg/files/11146127/build-arm64-osx-min1100-rel-config-37fce29a.log)

**Additional context**

The first failing build is https://github.com/microsoft/vcpkg/commit/37fce29a263a1546262eaed6af7ccac53b0094cd because of the command line parameter reordering a different parameter is eaten which cause the build error. 

See the attached logs. The expected output around line 13833 id 
```
qtrle_encoder=yes
r=yes
r10k_decoder=yes
```
in 47a020d we see: 
```
qtrle_encoder=yes
r=--debug
r10k_decoder=yes
```
and in failing 37fce29 it is 
```
qtrle_encoder=yes
r=--disable-x86asm
r10k_decoder=yes
```

It finally fails because of the missing --disable-x86asm on the arm64 target:  
```
In file included from src/libavutil/bswap.h:38:
src/libavutil/aarch64/bswap.h:38:13: error: invalid instruction mnemonic 'rev'
    __asm__("rev %w0, %w0" : "+r"(x));
            ^
<inline asm>:1:2: note: instantiated into assembly here
        rev %cx, %cx
        ^~~
```

--- a/ports/ffmpeg/build.sh.in
+++ b/ports/ffmpeg/build.sh.in
@@ -130,7 +130,7 @@ build_ffmpeg() {
 
 cd "$PATH_TO_BUILD_DIR"
 
-if [ $OSX_ARCH_COUNT -gt 1 ]; then
+if [ $OSX_ARCH_COUNT -gt 0 ]; then
     for ARCH in $OSX_ARCHS; do
         echo "=== CLEANING FOR $ARCH ==="
 

--- a/ports/ffmpeg/build.sh.in
+++ b/ports/ffmpeg/build.sh.in
@@ -136,7 +136,7 @@ if [ $OSX_ARCH_COUNT -gt 0 ]; then
 
         make clean && make distclean
 
-        build_ffmpeg $ARCH --enable-cross-compile --extra-cflags=-arch --extra-cflags=$ARCH --extra-ldflags=-arch --extra-ldflags=$ARCH
+        build_ffmpeg $ARCH --extra-cflags=-arch --extra-cflags=$ARCH --extra-ldflags=-arch --extra-ldflags=$ARCH
 
         echo "=== COLLECTING BINARIES FOR $ARCH ==="
 

--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -513,7 +513,7 @@ else()
     set(OPTIONS "${OPTIONS} --disable-libmfx")
 endif()
 
-set(OPTIONS_CROSS " --enable-cross-compile")
+set(OPTIONS_CROSS "--enable-cross-compile")
 
 # ffmpeg needs --cross-prefix option to use appropriate tools for cross-compiling.
 if(VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "([^\/]*-)gcc$")

--- a/ports/ffmpeg/vcpkg.json
+++ b/ports/ffmpeg/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ffmpeg",
   "version": "5.1.2",
-  "port-version": 5,
+  "port-version": 6,
   "description": [
     "a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.",
     "FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created. It supports the most obscure ancient formats up to the cutting edge. No matter if they were designed by some standards committee, the community or a corporation. It is also highly portable: FFmpeg compiles, runs, and passes our testing infrastructure FATE across Linux, Mac OS X, Microsoft Windows, the BSDs, Solaris, etc. under a wide variety of build environments, machine architectures, and configurations."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2450,7 +2450,7 @@
     },
     "ffmpeg": {
       "baseline": "5.1.2",
-      "port-version": 5
+      "port-version": 6
     },
     "ffnvcodec": {
       "baseline": "11.1.5.0",

--- a/versions/f-/ffmpeg.json
+++ b/versions/f-/ffmpeg.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a7e0c378cd0a6564b28c28d6d970e3d633b4382e",
+      "version": "5.1.2",
+      "port-version": 6
+    },
+    {
       "git-tree": "14577b12f56accfce4428caf17e4b47542f365ec",
       "version": "5.1.2",
       "port-version": 5

